### PR TITLE
Allow querying based on bounds parameter

### PR DESCRIPTION
**Pull request by whirish**
Example request: `http://localhost/api/territories/?bounds=((0.0,%200.0),%20(0.0,%2050.0),%20(50.0,%2050.0),%20(50.0,%200.0),%20(0.0,%200.0))`

Note that it must be parsed as a tuple of tuples (I think a dict with []s might work as well but I don't really care enough to test it), and to enforce a closed polygon it requires that the last tuple is the same as the first.

### CODE REVIEW

File: project/api/views.py [hunk 1]
```diff
@@ -20,7 +22,16 @@ class TerritoryViewSet(viewsets.ModelViewSet):
...
-    queryset = Territory.objects.all()
     serializer_class = TerritorySerializer
 
+    queryset = Territory.objects.all()
+
+    def get_queryset(self):
+        bounds = self.request.query_params.get('bounds', None)
+        if bounds is not None:
+            geom = Polygon(make_tuple(bounds), srid=4326)
+            self.queryset = Territory.objects.filter(geo__within=geom)
```
Comment (amaurymartiny) [inline project/api/views.py]: I didn't understand `within` in the docs https://docs.djangoproject.com/en/2.0/ref/contrib/gis/geoquerysets/. Isn't `bboverlaps` what we want? (No experience in this, just asking).

Comment (whirish) [reply project/api/views.py]: Yeah I think you're right actually, I don't have any experience in this either lol

**Comment by whirish** [comment]
Actually after a closer look it seems that the lookup just isn't working with any data, can't for the life of me find out why. No errors or anything or returned. Also after some closer examination I think `__intersects` lookup may be the best for our purposes as territories that are partially within the bounding box should be rendered as well.

**Comment by amaurymartiny** [comment]
"territories that are partially within the bounding box should be rendered as well" -> I think __bboverlaps does that too?

But I actually agree with you, let's go with __intersects first, it's more precise. Intuitively I'd guess that __intersects is heavier on resources than __bboverlaps, but we can optimize later (and I'm not even sure about what I'm saying).

**Pull request merged**